### ISSUE TITLE
News extension canonical problem

### ISSUE THREAD

**Issue by prentice87** [mblaschke/TYPO3-metaseo#269]
Hi,

I'm using metaseo and news 5.1.0 on t3 7.6.11.
On my blog detail page I can not get the correct canonical URL so I have to set "plugin.metaseo.metaTags.canonicalUrl = 0" in the template setup. If I don't it will only show the url up to the part where the blog title should appear and also this affect rx_shariff as it will only display the URL as well. Once canonical is switched off, rx_shariff works fine.

Is there a fix to this?

**Comment by thomaszbz** [comment]
I'm not aware that we had any users talking about rx_shariff yet. If you come up with more details or a patch, it could still be relevant to other users (together with other extensions than rx_shariff).

For the part of the news extension: Could you please make an example, together with some example output?

I assume you are using metaseo 2.0.0 release version?

**Comment by prentice87** [comment]
Typo3 7.6.11
Metaseo 2.0.0 beta
rx_shariff 7.3.1

I'm not really any good at extension development. Can you tell me what output you need? I'm glad to provide anything that helps. It's not a huge blog and we write everything ourserlves, so canonical isn't that important to us but Google might have a different opinion :)

If you need an HTML example, here is a link to a blog article from our blog: https://www.scansail.de/blog/artikel/das-beste-kommt-zum-schluss-segeln-im-herbst/

**Comment by thomaszbz** [comment]
Yes, the canonical seems important, especially in respect to duplicate content.

Now that the canonical tag is disabled in the link you provided, would you please post the output from the html source manually? To do so, you'd have to enable the canonical tag in metaseo temporarily.

**Comment by prentice87** [comment]
Okay, here it is
[blog.zip](https://github.com/mblaschke/TYPO3-metaseo/files/473226/blog.zip)

**Comment by thomaszbz** [comment]
Okay, then for the URL (generated with the news extension)

    https://www.scansail.de/blog/artikel/das-beste-kommt-zum-schluss-segeln-im-herbst/

you get the canonical output:

    <link rel="canonical" href="https://www.scansail.de/blog/">

But instead, you'd like the canonical tag to match with the URL:

    <link rel="canonical" href="https://www.scansail.de/blog/artikel/das-beste-kommt-zum-schluss-segeln-im-herbst/">

Right?

**Comment by prentice87** [comment]
Yes exactly. But also it somehow affects rx_shariff as that URL also changes to https://www.scansail.de/blog/
Somwhere where the URL is generated it must be shortend or set to just the page where the article is shown.

**Comment by thomaszbz** [comment]
Did the problem already occur in 7.6.10 and older? There's been a change in TYPO3 core, recently: #268

**Comment by prentice87** [comment]
Unfortunatly I don't know :/ Today I updated news to 5.1.0 and had to fix A LOT of issues because of that and that was one that was left. I already talked to georgringer but it seems this is not a news issue.
I just activated that option and the same keeps happening.

**Comment by prentice87** [comment]
Deactivating rx_shariff did not solve the problem btw.

**Comment by prentice87** [comment]
I hope there is a solution at some point, till then "mfc_canonical" creates the correct one and doesn't affect rx_shariff.

**Comment by thomaszbz** [comment]
Sounds that the rx_sharif needs the canonical, but seems, we're still missing support for the news extension. For the latter a duplicate of #234
